### ISSUE TITLE
Blurry on secondary monitor

### ISSUE THREAD

**Issue by thany** [Yutaka-Sawada/MultiPar#32]
I have two monitors:
* 27" 4K zoomed to 200%
* 19" SXGA zoomed to 100%

The program looks fine on my 4K monitor. On the other monitor, it's really blurry. Not because the resolution is lower, but definitely more blurry than it can be. Definitely more blurry than before I had my 4K screen.

Tried with the beta version as well - same deal.

Please note: you don't need a 4K monitor to test this. You just need one monitor at a different zoom level than the other.

**Comment by Yutaka-Sawada** [comment]
Thank you for bug report. The incident is same as [the previous one](https://github.com/Yutaka-Sawada/MultiPar/issues/19). Because I cannot test the behavior myself, I didn't not solve this issue. I made two samples of different application manifests. I'm not sure how it affects on dialog base application like MultiPar GUI. Please test them on your PC. 

I put the sample (gui_sample_2021-05-02.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

**Comment by JohnLGalt** [comment]
I have a 4K 144Hz monitor coming in soon, so I will have 2 monitors with very different sizes and scaling (currently, I use a pair of 1080p monitors @ 1920 * 1080), supposedly shipping this (coming) week sometime.  And, *technically* I have an older VGA monitor that has a much lower resolution than 1080p - I did bring it out but I haven't had the opportunity to connect it as I was working on 2 laptops with ChromeOS.  But I'm done with those, so I have space on my desk again.

Once I have them set up I'll post back and can test the issue much better.

**Comment by Yutaka-Sawada** [comment]
> Once I have them set up I'll post back and can test the issue much better.

I see. Thank you. The GUI consists of Microsoft Common Controls. There is little thing which I can do for this issue.

**Comment by JohnLGalt** [comment]
An app I use daily for file management, Xplorer², had some issues with monitors with different DPI settings.  He eventually added in support for that via a Multiple Monitor Manifest for the installer for X².

If you would like, I can talk to him (I already am on an email-conversation basis with him).

See this thread post https://forum.zabkat.com/viewtopic.php?t=11796 for more information on the Multiple Monitor manifest that he implemented.

**Comment by Yutaka-Sawada** [comment]
I failed to set dpiAwareness to be PerMonitorV2 on manifest. The setting of gdiScaling seems to over-write dpiAwareness or dpiAware item. From my test on primary monitor, gdiScaling looks better than Unaware (with old scaling method). But, dpiAware is always better than gdiScaling on primary monitor. I updated the sample and included screen-shots of each DPI percent. 

From [Microsoft's sample](https://github.com/microsoft/Windows-classic-samples/tree/master/Samples/DPIAwarenessPerWindow), PerMonitor dpiAwareness requires additional code to scale all GUI elements at start or WM_DPICHANGED. It's very complex. So, I feel that gdiScaling is simple and may be easier. 

I put the sample (gui_sample_2021-05-03.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

**Comment by Yutaka-Sawada** [comment]
I came up with a simple way to support different DPI monitors. It supports start-up only, and change font size at creating dialog. It doesn't support moving window between monitors. Because MultiPar GUI include a feature to change font size already, it's easy to implement a function to check monitor DPI and set proper font size. 

But, I don't know how it looks at starting on a secondary monitor. (Initial DPI at PerMonitor setting.) When Mr. thany has time, please take screen-shot on each monitor with new sample PerMonitorV2 version. In the PerMonitorV2 version, I didn't scale window yet. If it uses DPI of the secondary monitor, it should not be blurry, but may be different window size. It may be shown as 200% DPI (same as primary monitor) and big size, or be shown as 100% DPI and small original size. Before I make a scaling function, I need to know the original state.

**Comment by Yutaka-Sawada** [comment]
It seems that Windows 10 changes dialog size for the monitor's DPI automatically, when dpiAwareness is set to PerMonitorV2. But, the method isn't complete. While some controls are resized automatically, others are not. It's difficult to adjust them, while moving between monitors. So, I gave up to implement PerMonitorDPI method.

I prefer gdiScaling method for different DPI. But, it over-writes SystemDPI setting, and looks bad on a primary monitor. So, I made a function to switch scaling mode by DPI of monitors. When there is no difference between monitors's DPI, it adapts SystemDPI mode. When MultiPar's starting monitor's DPI is different from the primary monitor's DPI, it adapts gdiScaling mode. Though the gdiScaling is still blurry, it would be better than old scaling method. 

I put the sample (gui_sample_2021-05-07.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

**Comment by thany** [comment]
> 
> 
> Thank you for bug report. The incident is same as [the previous one](https://github.com/Yutaka-Sawada/MultiPar/issues/19). Because I cannot test the behavior myself, I didn't not solve this issue. I made two samples of different application manifests. I'm not sure how it affects on dialog base application like MultiPar GUI. Please test them on your PC.
> 
> I put the sample (gui_sample_2021-05-02.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

Sorry about that. 😔
There are a few applications with this problem. I did a cursory search and couldn't find an issue I created here.

Anyway, good to see this is beeing looked at 👍🏻

**Comment by Yutaka-Sawada** [comment]
Because Microsoft failed to design DPI per monitor method, other developers cannot solve this problem. Microsoft might gave up to implement also, as it introduced gdiScaling mode. It's not a perfect solution, but a possible better one. My implemented automatic switching DPI mode requires Windows 10 version 1703 or later. If it works enough (or a least acceptable level), I will enable this feature in the next version MultiPar 1.3.1.8.

**Comment by Yutaka-Sawada** [comment]
Because I don't have multiple monitors and cannot test myself, I cannot determine that this problem is possible to solve or not. There may be a way to fix. But, it requires help of a tester's effort. For example, I don't know what happens when moving a window between monitors. Though Windows OS seems to send some Window messages to the application, the kinds and order are unknown. So, I cannot write code for those Window messages for scaling task. 

By the way, from watching Windows Explorer 's behavior, it's not perfect. (It must be Microsoft's built-in OS tool.) Windows Explorer doesn't seem to follow the change of DPI dinamically. (I found this problem, while changeing DPI to check multiple size icons.) While the window basement and most components are scaled, some are not. It doesn't change icon size for new DPI. It doesn't scale dialog (property of file) correctly (looks blurry). (Restarting OS fixes these issues in Windows Explorer.) I don't know that it works between monitors. 

Anyway, I cannot solve this problem without testing the behavior. When Microsoft itslef doesn't fix a bug of Windows 10 's basic application, it would be more difficult for me.

**Comment by JohnLGalt** [comment]
The FedeEx tracking number shows delivery of the 4K monitor Monday, 17 May 2021.  I should have it, even with reasonable delays, but 19 May.

**Comment by JohnLGalt** [comment]
I got lucky.  The monitor was delivered today.  I have some things to do but I should be able to play with it a little tomorrow or Sunday to test this.

**Comment by Yutaka-Sawada** [comment]
Thanks John L Galt. I made a sample to see the effect of "dpiAwareness = PerMonitorV2". It's for test with Create window. Manual resize is disabled to see Windows OS's automatic resize on different DPI. When I check behavior by changing my monitor's DPI, it's difficult to pre-calculate exact window size of different DPI. When DPI is 120, result is 1 pixel different on my PC. But, this isn't so much problem. (Human being won't distingush the difference of a few pixels.) I think that openning MultiPar on a secondary monitor without blurry will be possible technically in future. Though it may have some small problems, it may work mostly. 

The most problem is how to add some functions to manual changing before/after Windows OS's automatic resizing. It needs to set window's min size before Windows OS's automatic resizing. (Without setting proper min size, OS cannot scale the window to smaller.) It needs to adjust/resize the dialog after Windows OS's automatic resizing. (A user may change window size manually.) To see order of window messages, the sample saves them on debug.txt. Please test it. I put the sample (gui_sample_2021-05-16.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

**Comment by JohnLGalt** [comment]
I'm unboxing the monitor today, so I'll grab this version and test with a few different versions and see if I can notice a difference.

**Comment by JohnLGalt** [comment]
OK, here come pictures.  Some strange things going on.

I use folder of 1.3.1.8 βeta that I created on 13 May 2021 (No other build info available from `About` button) and ran both Auto and PerMonitorV2 on 2 monitors, 1 with true 4K resolution and one with my usual 1080p resolution.


MP_Auto on 4K Monitor:

<img width="553" alt="MP 4k 05" src="https://user-images.githubusercontent.com/7432512/118557360-0f217c80-b733-11eb-9726-852c43579bc9.png">


MP_Auto on 1080p monitor:

<img width="368" alt="MP 4k 06" src="https://user-images.githubusercontent.com/7432512/118557464-35dfb300-b733-11eb-886d-0eba2a53c71f.png">


MP_PMV2 on 4K (this happens every time I open it):

<img width="912" alt="MP 4k 07" src="https://user-images.githubusercontent.com/7432512/118557514-4b54dd00-b733-11eb-89f7-7e1a59a64f43.png">


MP_PMV2 after manually resizing to normal dimensions:

<img width="578" alt="MP 4k 07 1" src="https://user-images.githubusercontent.com/7432512/118557587-5f98da00-b733-11eb-8ba9-a8dc50ca2720.png">


MP_PMV2 after dragging to older 1080p monitor (Note - it was not resizable *smaller*, only larger, on 1080p monitor):

<img width="581" alt="MP 4k 08" src="https://user-images.githubusercontent.com/7432512/118557692-89ea9780-b733-11eb-8ee1-610103ecdf3d.png">


There is still a little blurriness when moving from large to small window, and also note that Windows has set the 4K monitor to 150% Scaling for 4K monitor, and 100% for 1080p monitor.

OS:  Windows 10 21H1 (OS Build 19043.985)
HW:  https://valid.x86.fr/tf5prr

**Comment by Yutaka-Sawada** [comment]
> OK, here come pictures. Some strange things going on.

Thank you for test. By watching the screen-shot, gdiScaling feature seems to be enough for me. Though it's a little blurry still, it would be acceptable level in 150% DPI. It may depends on the scaling factor. If I cannot solve problem of PerMonitorV2 mode, I will use this method.

There are many problems in PerMonitorV2 mode. About the strange initial size, it might set wrong min size of window. I modified my sample to set 0 as min size. When I set my monitor's DPI to 150%, there is no problem. I don't know the case of Multiple monitors of different DPIs. Please test with new sample version. Also, please send "debug.txt" (made in a MultiPar.exe's directory) to see what happened. It writes initial window size and resize when. I put the sample (gui_sample_2021-05-18.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu).

**Comment by JohnLGalt** [comment]
Tested with newly downloaded Sample 05-18 extracted into folder of duplicated MP 1.3.1.8 as before.  System has been set up with Cleratype settings for clarity according to my own visual reference.

### MP_Auto on 4K Monitor:

<img width="912" alt="MP 4k 09" src="https://user-images.githubusercontent.com/7432512/118584894-de0e6f80-b765-11eb-9fb0-89e30207472c.png">


### MP_Auto dragged from 4K monitor to 1080p monitor:

<img width="608" alt="MP 4k 10" src="https://user-images.githubusercontent.com/7432512/118584930-f3839980-b765-11eb-9139-05cf79ea9869.png">


### MP_PMV2 on 4K Monitor:

<img width="578" alt="MP 4k 11" src="https://user-images.githubusercontent.com/7432512/118584964-fda59800-b765-11eb-9b5a-1694d3a2b574.png">


### MP_PMV2 dragged from 4K monitor to 1080p monitor:

<img width="368" alt="MP 4k 12" src="https://user-images.githubusercontent.com/7432512/118585005-101fd180-b766-11eb-93c6-7b638d1bf29d.png">

It looks a little better, but MP_Auto now was a much larger window than normal.  I think this is because of 150% scaling enabled?

**Comment by Yutaka-Sawada** [comment]
> Tested with newly downloaded Sample 05-18 extracted into folder of duplicated MP 1.3.1.8 as before.

Thank you for quick tests. From new screen-shot of PerMonitorV2 mode, setting minimum window size is important. It must set the min size before Windows OS's automatic scaling for different DPI. To do so, it needs to know the DPI change before actual scaling (auto resize window). While I see the series of window messages, WM_GETDPISCALEDSIZE may be good trigger. 

When I change DPI of my monitor, the order of messages seems to be like below;
WM_GETDPISCALEDSIZE  <- OS starts DPI change task.
WM_GETMINMAXINFO     <- OS refers min size of the window.
WM_SIZE              <- OS scales the window.
WM_DPICHANGED        <- OS finished DPI change task.

If this order is same as moving window between monitors, I can write code in the message process. These messages are written on "debug.txt". I made a sample for the messages. Please test with it. I put the sample (gui_sample_2021-05-18b.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu). 


> I think this is because of 150% scaling enabled?

No, it's because window size was changed ago. MultiPar_Auto.exe tries to resume the previous window size. Scaling itself works well. 

MultiPar_Auto.exe switches DPI mode by monitor's DPI. You might set primary monitor to 4K Monitor. When MultiPar's openned monitor's DPI is same as system DPI (primary monitor's DPI), it uses the system DPI without blur. Because your new screen-shot of "MP_auto on 4K Monitor" has no blur, I think that system DPI is 150%. Then, moving the window to another monitor gives blur like the screen-shot of "MP_Auto dragged from 4K monitor to 1080p monitor". 

When MultiPar's openned monitor's DPI is different from system DPI (primary monitor's DPI), it scales the window by gdiScaling method. Because your old screen-shot of "MP_auto on 4K Monitor" was blurry, I think that you moved the window from secondary monitor to primary monitor. This is a fault of auto switching DPI system, and it's not good at moving a window between monitors.

**Comment by JohnLGalt** [comment]
> No, it's because window size was changed ago. MultiPar_Auto.exe tries to resume the previous window size. Scaling itself works well.

Ahhh.  That actually makes sense to me.

OK, testing with new set right now.

Results in a few minutes.

**Comment by JohnLGalt** [comment]
Sorry, delayed by other stuff.  Testing now.

### V2 on 4K:

<img width="578" alt="MP V2 on 4k 13" src="https://user-images.githubusercontent.com/7432512/118730737-11a0d680-b806-11eb-9ea8-9e97c88146b7.png">


### V2 on 1080p (dragged over from 4K):

<img width="368" alt="MP V2 on 1080p 14" src="https://user-images.githubusercontent.com/7432512/118730763-1e252f00-b806-11eb-8836-78b3ee150321.png">

**Comment by Yutaka-Sawada** [comment]
> Sorry, delayed by other stuff. Testing now.

Thank you for test. It seems that dragging a window to another different DPI monitor gives same result as changing DPI of a monitor. But, the initial state is different. Because I did not read "debug.txt" yet, I don't know the actual changing steps. From my tests on my PC, the Create window can follow dynamic DPI change now. It will re-load icons and adjust child controls on the dialog. Though minimum size of the window has a little calculation error, it's ignorable. Nobody will care the difference of a few pixels. (Because each window has some padding space inner border, there is no controling problem.)

I'm not sure it works with moving window between monitors, or openning MultiPar on a secondary monitor. Please test these cases. I put the sample (gui_sample_2021-05-19.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu). If it works, I will disable printing out of "debug.txt". If it doesn't work, send the "debug.txt" file, so that I can see what happened.

**Comment by JohnLGalt** [comment]
Ok, I'll test in a few minutes.

**Comment by JohnLGalt** [comment]
Something weird here.

### V2 Start on 4K:

<img width="579" alt="MP V2 open on 4k 15" src="https://user-images.githubusercontent.com/7432512/118752662-850c0d80-b831-11eb-858f-592d9f9d2448.png">


### Above window dragged to 1080p monitor:

<img width="369" alt="MP V2 open on 4k dragged to 1080p 16" src="https://user-images.githubusercontent.com/7432512/118752690-93f2c000-b831-11eb-8d57-eee7f3e7bc94.png">


### Above windows closed, then MP run again, opens on 1080p monitor:

<img width="368" alt="MP V2 open on 1080p 17" src="https://user-images.githubusercontent.com/7432512/118752728-a3720900-b831-11eb-8cac-4af184f92d59.png">


### Above windows dragged to 4K monitor:

<img width="578" alt="MP V2 open on 1080p Dragged to 4K 18" src="https://user-images.githubusercontent.com/7432512/118752752-acfb7100-b831-11eb-8641-c975c74a16c7.png">

### 
Above windows closed, and MP run again (so open on 4K monitor once again):

<img width="578" alt="MP V2 re-opened on 4k 19" src="https://user-images.githubusercontent.com/7432512/118752785-bab0f680-b831-11eb-8557-39ab4f610ad1.png">

**Comment by Yutaka-Sawada** [comment]
> Something weird here.

Yes, the steps of openning window on secondary monitor seems to be different. I don't know what happened without "debug.txt". I cannot write proper code now. I will try in future, when someone tests. Thanks John Galt for many tests.

**Comment by JohnLGalt** [comment]
Oh, sorry, you wanted a copy of the debug.txt - I totally forgot to upload it.

Here it is.

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6509941/debug.txt)

**Comment by JohnLGalt** [comment]
I kept each test in a separate folder, and have not deleted any folders yet.  I can upload them all.

I'll link to the tests I conducted above, then upload the appropriate debug.txt for each.

### First test:  

https://github.com/Yutaka-Sawada/MultiPar/issues/32#issuecomment-842647479

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6509962/debug.txt)


### Second Test:

https://github.com/Yutaka-Sawada/MultiPar/issues/32#issuecomment-842794700

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6509968/debug.txt)


### Third Test:

https://github.com/Yutaka-Sawada/MultiPar/issues/32#issuecomment-843607506

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6509976/debug.txt)


### Fourth Test:

https://github.com/Yutaka-Sawada/MultiPar/issues/32#issuecomment-843720517

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6509978/debug.txt)


That's all of them.

To make it easier, I'll now also go back and edit each test post and upload the relevant debug.txt inside them, so if you want to look at post directly, debut.txt will also be there.

Again, my apologies, I completely forgot about uploading these files, and I apologize.

**Comment by Yutaka-Sawada** [comment]
> Oh, sorry, you wanted a copy of the debug.txt - I totally forgot to upload it.
> Again, my apologies, I completely forgot about uploading these files, and I apologize.

There is no problem. Because you don't owe testing nor it's not obligation, no need to apologize. 

From reading the "debug.txt", Windows OS doesn't send WM_GETDPISCALEDSIZE message before changing window size at first. Maybe it's because the window was not shown yet. I improved code for a case of opening position monitor DPI is different from system (primary monitor) DPI. As it's complex and depends on order of incomming messages, I'm not sure that it works on all users cases. I put the sample (gui_sample_2021-05-20.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu). 

In this time, please check icon, text and column size, by adding source file on file-list. I might forget to adjust the window's child controls, while Windows OS updates most of them. If it works well, I need to change all other dialogs in same way.

**Comment by JohnLGalt** [comment]
No, you did ask for debug.txt in one of your posts, and I completely ignored it and only posted ss.

I'm reading more carefully now ;p

I'll be taking screenshots here and posting them and debug after - maybe 30 minute or so.

**Comment by JohnLGalt** [comment]
Good thing you asked about loading parity file.


### 1.  Opened new V2 on 1080p:

<img width="368" alt="MP DPI Testing 5 - 01 - Inital open on 1080p monitor" src="https://user-images.githubusercontent.com/7432512/118913510-9cadc980-b8f7-11eb-8fd2-f1d8ad6c02ce.png">


### 2.  Dragged to 4K:

<img width="578" alt="MP DPI Testing 5 - 02 - Dragged window to 4K" src="https://user-images.githubusercontent.com/7432512/118913540-a7685e80-b8f7-11eb-85d6-fac8d12ba1e6.png">


### 3.  Dragged back to 1080p, opened parity to verify integrity:

<img width="761" alt="MP DPI Testing 5 - 03 - Added  par file for integrity check" src="https://user-images.githubusercontent.com/7432512/118913561-b3542080-b8f7-11eb-8f32-94bf0e674845.png">


### 4.  Dragged integrity check to 4K:

<img width="1196" alt="MP DPI Testing 5 - 04 - Dragged verification window to 4K" src="https://user-images.githubusercontent.com/7432512/118913587-bf3fe280-b8f7-11eb-81ae-0b8ae3a78d8c.png">


### 5.  Closed integrity check, then opened V2 again (so auto opens on 4K):

<img width="578" alt="MP DPI Testing 5 - 05 - Inital window opened on 4K" src="https://user-images.githubusercontent.com/7432512/118913624-ce269500-b8f7-11eb-8112-6a63ed59eba0.png">


### 6.  Dragged to 1080p

<img width="368" alt="MP DPI Testing 5 - 06 - Dragged initial window to 1080p" src="https://user-images.githubusercontent.com/7432512/118913632-d4b50c80-b8f7-11eb-8aee-40c01905c78c.png">


### 7.  Dragged back to 4K, added parity file to check integrity:

<img width="1196" alt="MP DPI Testing 5 - 07 - Opened parity file for integrity check" src="https://user-images.githubusercontent.com/7432512/118913656-df6fa180-b8f7-11eb-8ffb-e9d8b903a98d.png">


### 8.  Dragged Integrity verification window back to 1080p:

<img width="761" alt="MP DPI Testing 5 - 08 - Dragged integrity check window to 1080p" src="https://user-images.githubusercontent.com/7432512/118913683-ef878100-b8f7-11eb-8a25-bd83942c2b7b.png">

### 
Debug log:

[debug.txt](https://github.com/Yutaka-Sawada/MultiPar/files/6512612/debug.txt)


(**_Note**:  It will be a little out of order, because I did #1, the n 3, then 4, then closed 4 and did 1 again and then 2, then closed 2 and did 5-8_).

**Comment by Yutaka-Sawada** [comment]
Thanks JohnLGalt for tests. Opening "Create window" on a secondary monitor seems to work well now. There is no blur. I wrote same method for "Verify window" and "Recreate window". Though It will work, I might forget to update something. Please test each case. If it works well, no need to post screen-shot. If there is a wrong size, position, or blurry, please report it.

Now, how to treat another child window is a problem. They are "Creating window", "About window", "Option window", "Preview window", and "Comment window". I don't know how it looks on a secondary monitor of different DPI. If it inherits DPI of the parent window, OS may resize samely. Another problem is a case of putting windows on different monitors. It's difficult to support this case. At this time, using MultiPar on a same monitor is good. While moving the main (Cretae or Verify) window between monitors is possible, its child window should exist on the same monitor as the parent window.

I put the sample (gui_sample_2021-05-21.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu). There is no restriction or debug feature in this version. You can use this for normal usage, but it may have a problem still. For example, the behavior of a child window is unknown.

**Comment by JohnLGalt** [comment]
Just so I am clear:

I should perform all tests (pretty much use MP as normal, verifying integrity of existing parity sets, creating parity sets of new set of files, verify setup, running batch commands, and anything else that I would normally do and only post if I have problems using latest Sample?

OK, but I think I will enable debugging and run it in a test folder and verify no issues, if I can enable debugging, and if everything works well I'll just report back that I had no issues, and then replace my stable or βeta portable and start using it as default.

**Comment by Yutaka-Sawada** [comment]
Because I forgot to adjust icon of "Creating window" for current DPI, I updated the sample. I put the sample (gui_sample_2021-05-22.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOaSo1n_R8awJ_hg0?e=4V0gXu). The difference is just a GUI's visual. When there is only one monitor, or their DPI are same between multiple monitors, there will be no problem at all. This sample is for those who uses Windows 10 and multiple monitors of different DPI. 

If Mr. thany has time to test, please try the new sample version. The environment is varied by users. If there is no serious problem, I will release next beta version 1.3.1.8. So that, more users may see the behavior on their PC. I will fix a problem in rare case later.

**Comment by JohnLGalt** [comment]
I did everything I could to try to break it from moving windows from one monitor to another to creating comments, creating parity archive st, verifying existing set, opening options, clicking any about button I saw in any window, and more, and when paired with the DLLs and setup files from 1.3.1.8 β I had 0 issues.

The only time anything breaks is is you leave it partially on both monitors, so that the majority is on the larger monitor, and the part on the smaller monitor is sized too largely - but this happens with **every** application, and there is nothing that can be done about it, and it is such a small niche case (I'm pretty sure 99.99% of people will not have application 'straddling' 2 monitors, but rather will have any given app on one monitor or the other), so I think 1.3.1.8 is ready for prime time.

**Comment by Yutaka-Sawada** [comment]
> I think 1.3.1.8 is ready for prime time.

Thanks John L. Galt for many tests sometimes. I released v1.3.1.8 today.

**Comment by Miniland1333** [comment]
Has this issue been completely solved and therefore closed successfully?

**Comment by Yutaka-Sawada** [comment]
> Has this issue been completely solved and therefore closed successfully?

I don't know, because I cannot test the behavior myself. Then, I didn't close this issue yet.

**Comment by JohnLGalt** [comment]
> Has this issue been completely solved and therefore closed successfully?

With all the testing I performed, through 1.3.1.8 release, the only issue left is if you try to have the Multi-Par window occupy space on 2 differently sized monitors with different DPI settings - aka straddling them both, and particularly when the majority of MP is on the monitor with the higher DPI. This causes the part that is on the lower DPI to scale badly - but this happens with every application that autoscales that I tried, so it is not something that should be a factor, Everything else worked flawlessly.

I'm now using a trio of 4K monitors with identical DPI, so I no longer even see that one straddling issue, but I can dig out a different monitor with different resolution and DPI and see if anything has reverted since. Otherwise, I'd call it solved.
